aiobotocore/httpxsession.py

```diff
@@ -0,0 +1,305 @@
+from __future__ import annotations
+
+import asyncio
+import io
+import os
+import socket
+from typing import TYPE_CHECKING, Any, cast
+
+import botocore
+from botocore.awsrequest import AWSPreparedRequest
+from botocore.httpsession import (
+    MAX_POOL_CONNECTIONS,
+    EndpointConnectionError,
+    HTTPClientError,
+    ReadTimeoutError,
+    create_urllib3_context,
+    ensure_boolean,
+    get_cert_path,
+    logger,
+)
+from multidict import CIMultiDict
+
+import aiobotocore.awsrequest
+from aiobotocore._endpoint_helpers import _text
+
+# TODO: resolve future annotations thing
+
+
+try:
+    import httpx
+except ImportError:
+    httpx = None
+
+if TYPE_CHECKING:
+    from ssl import SSLContext
+
+
+class HttpxSession:
+    def __init__(
+        self,
+        verify: bool = True,
+        proxies: dict[str, str] | None = None,  # {scheme: url}
+        timeout: float | list[float] | tuple[float, float] | None = None,
+        max_pool_connections: int = MAX_POOL_CONNECTIONS,
+        socket_options: list[Any] | None = None,
+        client_cert: str | tuple[str, str] | None = None,
+        proxies_config: dict[str, str] | None = None,
+        connector_args: dict[str, Any] | None = None,
+    ):
+        if httpx is None:  # pragma: no cover
+            raise RuntimeError(
+                "Using HttpxSession requires httpx to be installed"
+            )
+        if proxies or proxies_config:
+            raise NotImplementedError(
+                "Proxy support not implemented with httpx as backend."
+            )
+
+        # TODO: handle socket_options
```

Comment:
should we error if anything set then until it's handled?

Reply:
oh lol this `TODO` is copy-pasted from `AIOHTTPSession`. So whatever we consider doing here should be mirrored there. I propose it be handled in a different issue/PR